test/test_save.rb

```diff
@@ -8,21 +8,22 @@ class TestSave < Test::Unit::TestCase #:nodoc:
   # games to be initialised with different options.
 
   def test_save_crystal
-    game = ScottKit::Game.new({ :random_seed => 12368, :echo_input => true })
+    game = ScottKit::Game.new(random_seed: 12368, echo_input: true,
+                              output: StringIO.new)
     game.load(IO.read("games/test/crystal.sao"))
-    withIO(File.new("games/test/crystal.save-script"), 
-           File.new("/dev/null", "w")) do
+    withIO(File.new("games/test/crystal.save-script"), $stdout) do
       game.play()
     end
     assert_equal(File.read("TMP"), File.read("games/test/crystal.save-file"))
     File.unlink "TMP"
   end
 
   def test_resave_crystal
-    game = ScottKit::Game.new({ :random_seed => 12368, :echo_input => true,
-        :restore_file => "games/test/crystal.save-file" })
+    game = ScottKit::Game.new(random_seed: 12368, echo_input: true,
+                              output: StringIO.new,
+                              restore_file: "games/test/crystal.save-file")
     game.load(IO.read("games/test/crystal.sao"))
-    withIO(StringIO.new("save game\nTMP"), File.new("/dev/null", "w")) do
+    withIO(StringIO.new("save game\nTMP"), $stdout) do
```

Comment:
Why are you no longer discarding the output of `play` here? We don't want that output, just the save file that is generated along the way.

Reply:
This change I could probably back out. I'd started out trying to see if I could totally remove the need for `withIO` but left it here to pass in the save game command. Since `withIO` requires both params so I had to pass something in... the output is still being discarded though via the output string instead of `/dev/null`. 

I think we probably could get rid of `withIO` here by just loading the game file into a StringIO object and then appending `"save game\nTMP"`. I'll give it a try and see if that end up looking cleaner.

Reply:
I don't see how writing the output of `play` to `$stdout` can discard it -- surely it will appear on standard output?

(Probably the Ruby library provides a null-file object somewhere, which exists to discard output; that would really be the best thing to use here.)

You know, it's kind of cool that something as trivial as `withIO` lets you do so many things :-)

Reply:
there's nothing going to `$stdout` because it's getting sent to the `:output` stream instead. `$stdout` is only used when nothing is provided. another option would be to modify `withIO` to only assign non-nil values.